internal/provider/physical_replication_stream.go

```diff
@@ -0,0 +1,384 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/cockroachdb/cockroach-cloud-sdk-go/v6/pkg/client"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+type PhysicalReplicationStream struct {
+	PrimaryClusterId      types.String `tfsdk:"primary_cluster_id"`
+	StandbyClusterId      types.String `tfsdk:"standby_cluster_id"`
+	ID                    types.String `tfsdk:"id"`
+	CreatedAt             types.String `tfsdk:"created_at"`
+	ReplicatedTime        types.String `tfsdk:"replicated_time"`
+	ReplicationLagSeconds types.Int64  `tfsdk:"replication_lag_seconds"`
+	RetainedTime          types.String `tfsdk:"retained_time"`
+	Status                types.String `tfsdk:"status"`
+	FailoverAt            types.String `tfsdk:"failover_at"`
+	FailoverImmediately   types.Bool   `tfsdk:"failover_immediately"`
+	ActivatedAt           types.String `tfsdk:"activated_at"`
+}
```

Comment:
Do you mind moving this to the internal/provider/models.go file?   

I wouldn't actually mind having them in their specific resource files but for now, the convention is to have them there.